runtime/integration-tests/src/generic/utils/genesis.rs

```diff
@@ -18,3 +32,117 @@ impl Genesis {
 		self.storage
 	}
 }
+
+pub fn balances<T: Runtime>(balance: Balance) -> impl GenesisBuild<T> {
+	pallet_balances::GenesisConfig::<T> {
+		balances: default_accounts()
+			.into_iter()
+			.map(|keyring| (keyring.id(), balance))
+			.collect(),
+	}
+}
+
+pub fn tokens<T: Runtime>(values: Vec<(CurrencyId, Balance)>) -> impl GenesisBuild<T> {
+	orml_tokens::GenesisConfig::<T> {
+		balances: default_accounts()
+			.into_iter()
+			.map(|keyring| {
+				values
+					.clone()
+					.into_iter()
+					.map(|(curency_id, balance)| (keyring.id(), curency_id, balance))
+					.collect::<Vec<_>>()
+			})
+			.flatten()
+			.collect(),
+	}
+}
+
+pub fn assets<T: Runtime>(currency_ids: Vec<CurrencyId>) -> impl GenesisBuild<T> {
+	orml_asset_registry::GenesisConfig::<T> {
+		assets: currency_ids
+			.into_iter()
+			.map(|currency_id| (currency_id, currency::find_metadata(currency_id).encode()))
+			.collect(),
+		last_asset_id: Default::default(), // It seems deprecated
+	}
```

Comment:
Why redo what was already in the non generic part?

Reply:
If you mean, why not reuse the genesis functions from `utils/genesis.rs` is because:
- I've tried, based on them, to make them more generic. The genesis methods from that module assume some currencies and initial values that maybe are not needed for some tests. These implementations are a bit more generic. Ideally, I would need to modify the others but doing it would make a lot of tests fail. Out of the scope of this PR, but we'll end up unifying them.
- Nit. Returning `impl GenesisBuild<T>` improves the ergonomics with the `Genesis<T>` type (added in the previous PR). Passing a mutable ref to storage and later assimilating is a bit awkward, not sure why Substrate has chosen this.

Reply:
i.e: with this:
```rust
Genesis::<T>::default()
   .add(genesis::balances(T::ExistentialDeposit::get() + FOR_FEES))
   .add(genesis::assets(vec![Usd6::ID]))
   .add(genesis::tokens(vec![(Usd6::ID, Usd6::ED)]))
   .storage()
```
You know exactly how your initial state is for your current test without looking into the genesis functions. No more balances that can you see, nor more assets.

Reply:
You can always wrap a common initial state under some method and then call this method for a set of tests that requires that specific state. In the loan example, the common state is a function intended to be called from different loan tests that need the same pool state.

It's a very low-level foundation

Reply:
Actually, each test domain could implement a `GenesisBuild` where its initial state is built using the previous pieces, and later:

```rust
Genesis::<T>::default()
   .add(genesis::loans_initial_state())
   .storage()
```

or 

```rust
Genesis::<T>::default()
   .add(genesis::empty_pool())
   .storage()
```

Things like those. Just drafting an idea of how it can evolve

Reply:
My point in favor of this is that I feel quite frustrated when a test I didn't write fails, and I need to dig into the first calls to know precisely what was the initial state it was built. And it was because of some global thing I didn't consider that was related to the failure.